# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/ipc/ContentParent.cpp
# Commit: f8541a07da37
# Full Hash: f8541a07da376ccd25baa5acaaed9220fbf24fad
# Author: Andrew McCreight <continuation@gmail.com>
# Date: 2022-08-12 09:37:14
# Description:
#   Bug 1784036 - Don't reject null principals in RecvCompleteAllowAccessFor. r=timhuang
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D154396
# ==============================================================================

diff -r 27f856541c9f -r f8541a07da37 dom/ipc/ContentParent.cpp
--- a/dom/ipc/ContentParent.cpp	Thu Aug 11 17:39:53 2022 +0000
+++ b/dom/ipc/ContentParent.cpp	Thu Aug 11 17:42:07 2022 +0000
@@ -6609,10 +6609,6 @@
     return IPC_OK();
   }
 
-  if (!aTrackingPrincipal) {
-    return IPC_FAIL(this, "No principal");
-  }
-
   StorageAccessAPIHelper::CompleteAllowAccessFor(
       aParentContext.get_canonical(), aTopLevelWindowId, aTrackingPrincipal,
       aTrackingOrigin, aCookieBehavior, aReason, nullptr)
